src/helpers/parallel.ml

```diff
@@ -0,0 +1,26 @@
+open Lwt_ext.Let_syntax
+
+(* lazy because Unix.fork stops working after domains start *)
+(* TODO: proper number of domains *)
+(* THIS IS GLOBAL STATE OUTSIDE OF src/node/server.ml. CAUTION *)
+let pool = lazy (Lwt_domain.setup_pool 8)
+let pool () = Lazy.force_val pool
```

Comment:
Shouldn't we avoid this on account that Lazy.force_val is racy? Is it ok just because `setup_pool` is fast?

Reply:
What race do you have in mind? I'm pretty sure `Lazy.t` is just a more efficient wrapper over what we're doing elsewhere, which is `'a option ref`. If Lazy.t is subject to race's, then so is our ref, and then we have a big issue with async state hashing.

Reply:
cc @EduardoRFS

Reply:
Answer from @EduardoRFS: 
- Arthur is right that Lazy.t is not thread safe.
- But here the Lwt_domain pool is only accessed in the main thread, so it's ok (just like the task pool for async state hashing is only accessed from the main thread).
- If we spawned domains from inside domains, this would not be safe, but we don't ,so it's ok.

Reply:
Ah yes, sure. Thanks!